### PR TITLE
sorbet-runtime: Better error for params misuse

### LINKED ISSUE #3328: A better error message for invalid signatures, when using only sorbet-runtime
### Problem

The following signature is invalid, but it's easy to see how someone could make this mistake.

```
sig { params(Integer).void }
def f(i)
```

The correct signature, of course, is `params(i: Integer)`.

The error message produced by `sorbet-runtime` is quite poor:

```
undefined method `keys' for Integer:Class (NoMethodError)
```

### Proposed solution

The error message should instead say something like: "Invalid signature: params expects a Hash"

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Fixes #3328

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
cc @jaredbeck

File: gems/sorbet-runtime/test/types/builder_syntax.rb [hunk 1]
```diff
@@ -29,6 +29,17 @@ def self.fn(x); x.to_s; end
...
+    it 'requires paramms be keyword args' do
+      ex = assert_raises do
+        Class.new do
+          extend T::Sig
+          sig {params(Integer).void}
+          def self.foo; end; foo
+        end
+      end
+      assert_includes(ex.message, "params expects keyword arguments")
+    end
```
Comment (elliottt) [inline gems/sorbet-runtime/test/types/builder_syntax.rb:41]: Could you add a case for `sig {params.void}` as well?

Comment (jez) [reply gems/sorbet-runtime/test/types/builder_syntax.rb:41]: https://github.com/sorbet/sorbet/pull/3524/files#diff-3f67f2cd0e5ede62a49719cac03477abf49e2fa975343aa21859a58c8d293dc4R43

**Review by elliottt — APPROVED** [review]
🚢

**Pull request merged**